GliaWidgets/Lib/Download/FileDownload.swift

```diff
@@ -33,10 +33,8 @@ class FileDownload {
     let file: ChatEngagementFile
 
     private var storageID: String? {
-        if let fileName = file.name {
-            return fileName
-        } else if let fileID = file.id {
-            return fileID
+        if let fileID = file.id, let fileName = file.name {
+            return fileID + "CODE:" + fileName
```

Comment:
Let's use another separator. First of all, it would look better. Second, iOS doesn't allow for a semicolon in a filename (if you wish to save it to a device).
I suggest using `#` (or even better just `_`). The fileID seems to be just lowercase letters, numbers and dashes, so you won't have any interference if you'll just search for the first occurrence of this symbol to split the name.